addons/payment_ogone/models/payment_transaction.py

```diff
@@ -0,0 +1,258 @@
+# -*- coding: utf-'8' "-*-"
+
+import datetime
+import logging
+from lxml import etree, objectify
+from pprint import pformat
+import time
+from urllib import urlencode
+import urllib2
+
+from openerp import _, api, models
+from openerp.addons.payment.models.payment_acquirer import ValidationError
+from openerp.addons.payment_ogone.data import ogone_errors
+from openerp.tools import DEFAULT_SERVER_DATE_FORMAT
+from openerp.tools.float_utils import float_compare
+from openerp.tools.safe_eval import safe_eval
+
+_logger = logging.getLogger(__name__)
+
+
+class PaymentTransaction(models.Model):
+    _inherit = 'payment.transaction'
+    # ogone status
+    _ogone_valid_tx_status = [5, 9]
+    _ogone_wait_tx_status = [41, 50, 51, 52, 55, 56, 91, 92, 99]
+    _ogone_pending_tx_status = [46]   # 3DS HTML response
+    _ogone_cancel_tx_status = [1]
+
+    # --------------------------------------------------
+    # FORM RELATED METHODS
+    # --------------------------------------------------
+
+    @api.model
+    def _ogone_form_get_tx_from_data(self, data):
+        """ Given a data dict coming from ogone, verify it and find the related
+        transaction record. Create a payment method if an alias is returned."""
+        reference, pay_id, shasign, alias = data.get('orderID'), data.get('PAYID'), data.get('SHASIGN'), data.get('ALIAS')
+        if not reference or not pay_id or not shasign:
+            error_msg = _('Ogone: received data with missing reference (%s) or pay_id (%s) or shashign (%s)') % (reference, pay_id, shasign)
+            _logger.info(error_msg)
+            raise ValidationError(error_msg)
+
+        # find tx -> @TDENOTE use paytid ?
+        tx = self.search([('reference', '=', reference)], limit=1)
```

Comment:
tx --> transaction,
don't use limit=1, otherwise one of below conditions would never be satisfied.